tests/test_pyi_files.py

```diff
@@ -1,12 +1,45 @@
 import glob
 import os
+import os.path
 import re
 import subprocess
 import sys
 from itertools import zip_longest
 
 import pytest
 
+IGNORED_DEPRECATION_WARNINGS = [
+    # Ignore all DeprecationWarnings that come from pyflakes or flake8-bugbear
+    rf"{re.escape(os.path.join('pyflakes', 'checker.py'))}:\d+: DeprecationWarning: ",
+    r"bugbear\.py:\d+: DeprecationWarning: ",
+]
+IGNORED_DEPRECATION_PATTERNS = [
+    re.compile(pattern) for pattern in IGNORED_DEPRECATION_WARNINGS
+]
+
+
+def get_filtered_stderr(stderr: str) -> str:
+    lines = stderr.splitlines()
+
+    grouped_lines = []
+    skip_this_line = False
+    for line, next_line in zip_longest(lines, lines[1:]):
```

Comment:
This seems like it would be easier with a `re.sub(some_regex, "", stderr)` that removes the following line as well.

Reply:
Actually, I don't think we need to filter the output at all. Python itself can do the filtering. In fact we don't event need to capture the output, because it is possible to just turn warnings into errors.

To test this, I made a file named `foo.py` that contains `import imp` (a deprecated module). Then I ran python like this:

```
python3 -Werror::DeprecationWarning -Wignore::DeprecationWarning:foo
```

Now `import imp` fails but `import foo` succeeds:

```
>>> import imp
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.9/imp.py", line 31, in <module>
    warnings.warn("the imp module is deprecated in favour of importlib; "
DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
>>> import foo
>>> 
```

Reply:
> To test this, I made a file named `foo.py` that contains `import imp` (a deprecated module). Then I ran python like this:
> 
> ```
> python3 -Werror::DeprecationWarning -Wignore::DeprecationWarning:foo
> ```
> 
> Now `import imp` fails but `import foo` succeeds:

That seems fragile. I can reproduce the traceback if `imp` is imported before `foo`:

```pycon
>python -We::DeprecationWarning -Wignore::DeprecationWarning:foo
Python 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import imp
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\alexw\AppData\Local\Programs\Python\Python311\Lib\imp.py", line 31, in <module>
    warnings.warn("the imp module is deprecated in favour of importlib and slated "
DeprecationWarning: the imp module is deprecated in favour of importlib and slated for removal in Python 3.12; see the module's documentation for alternative uses
>>> import foo
>>>
```

But if I import `foo` before `imp`, no traceback is raised (and indeed, the DeprecationWarning is completely silenced):

```pycon
>python -We::DeprecationWarning -Wignore::DeprecationWarning:foo
Python 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import foo
>>> import imp
>>>
```

Reply:
I think ^that fragility may just be a weirdness of the REPL though. It seems to work find outside the REPL.

I knew there had to be a simpler way of doing this, thanks! :)